src/momento/aio/_add_header_client_interceptor.py

```diff
@@ -32,13 +32,51 @@ async def intercept_unary_unary(
         client_call_details: grpc.aio._interceptor.ClientCallDetails,
         request: grpc.aio._typing.RequestType,
     ) -> Union[grpc.aio._call.UnaryUnaryCall, grpc.aio._typing.ResponseType]:
+
+        # FIXME Hack to account for ddtrace lib injecting grpc metadata as list instead of proper grpc.aio.Metadata()
+        # Will open issue on data dog repo to discuss this more.
+        # Since once client_call_details.metadata is set as list we cant overwrite with `Metadata()` class we build
+        # up a new ClientCallDetails() in each request now in this interceptor and make sure Metadata() is set properly
+        new_client_call_details = None
+
+        # If no metadata set on passed in client call details then we are first to set, so we should just initialize
         if client_call_details.metadata is None:
-            client_call_details.metadata = Metadata()
+            new_client_call_details = ClientCallDetails(
+                method=client_call_details.method,
+                timeout=client_call_details.timeout,
+                metadata=Metadata(),
+                credentials=client_call_details.credentials,
+                wait_for_ready=client_call_details.wait_for_ready,
+            )
+
+        # This is block hit when ddtrace interceptor runs first and sets metadata as a list
+        elif isinstance(client_call_details.metadata, list):
+            existing_headers = client_call_details.metadata
+            metadata = Metadata()
```

Comment:
Because they passed it as a `list` in this case in `client_call_details.metadata` im making a net new `gprc.aio.Metadata()` object to build up with the values from their `list`

Reply:
okay, i see.  i thought they were creating a `list` with a single `Metadata` in it.